hazelcast/src/main/java/com/hazelcast/internal/management/operation/ScriptExecutorOperation.java

```diff
@@ -96,14 +83,8 @@ protected void writeInternal(ObjectDataOutput out) throws IOException {
     protected void readInternal(ObjectDataInput in) throws IOException {
         engineName = in.readUTF();
         script = in.readUTF();
-        int size = in.readInt();
-        if (size > 0) {
-            bindings = createHashMap(size);
-            for (int i = 0; i < size; i++) {
-                String key = in.readUTF();
-                Object value = in.readObject();
-                bindings.put(key, value);
-            }
+        if (in.getVersion().isLessThan(V3_10)) {
```

Comment:
Same here.

Reply:
If the class is not `Versioned`, it means it does not write version info in the stream and the target member will not have the version info (it will be `UNKNOWN`). More specifically, if a class does not implement `Versioned` in 3.9 but it does in 3.10, this means that a 3.10 member will receive the serialized format with version `UNKNOWN` from a 3.9 member. Thus, only these checks are actually safe in 3.10:
`if (out.getVersion().isUnknownOrLessThan(V3_10))`
`!if (out.getVersion().isGreaterOrEqual(V3_10))`
The `writeObject` method may keep the `out.getVersion().isLessThan(V3_10)` since it gets the version set by the serialization service but for the sake of clarity, it's best to keep the same check in `writeObject` and `readObject`

Reply:
Thanks for the comments @mmedenjak and @Donnerbart.  I'll update both checks to include unknown.